classes/manager.php

```diff
@@ -133,5 +133,23 @@ public static function get_total_weight() {
         return $totalweight;
     }
 
+    /**
+     * Checks that provided factorid exists and belongs to current user.
+     *
+     * @param string $factorname
+     * @param int $factorid
+     * @return bool
+     * @throws \dml_exception
+     */
+    public static function is_factorid_valid($factorname, $factorid) {
```

Comment:
Can you please pull $USER out and do that as a function param instead

This will make it easier to unit test, and also better later for when we do non-user actions, eg an admin is revoking it